arlo-client/src/components/Audit/Setup/Participants/useJurisdictionManifest.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'react'
+import { toast } from 'react-toastify'
+import { api } from '../../../utilities'
+import { IFileInfo } from '../../useJurisdictions'
+
+const loadJurisdictionManifest = async (
+  electionId: string
+): Promise<IFileInfo | null> => {
+  try {
+    return await api(`/election/${electionId}/jurisdiction/file`)
+  } catch (err) {
+    toast.error(err.message)
+    return null
+  }
+}
+
+const putJurisdictionManifestFile = async (
+  electionId: string,
+  csv: File
+): Promise<boolean> => {
+  const formData: FormData = new FormData()
+  formData.append('jurisdictions', csv, csv.name)
+  try {
+    await api(`/election/${electionId}/jurisdiction/file`, {
+      method: 'PUT',
+      body: formData,
+    })
+    return true
+  } catch (err) {
+    toast.error(err.message)
+    return false
+  }
+}
+
+const useJurisdictionManifest = (
+  electionId: string
+): [IFileInfo | null, (csv: File) => Promise<boolean>] => {
+  const [
+    jurisdictionManifest,
+    setJurisdictionManifest,
+  ] = useState<IFileInfo | null>(null)
+
+  const uploadJurisdictionManifest = async (csv: File): Promise<boolean> => {
+    // TODO poll for result of upload
+    if (await putJurisdictionManifestFile(electionId, csv)) {
+      setJurisdictionManifest(await loadJurisdictionManifest(electionId))
```

Comment:
what's the reasoning for round-tripping through server here? Can we just set the local state to `csv`?

Reply:
@jonahkagan Thoughts? I just literally copied this from the ballot manifest hook, and changed names. :grin:

Reply:
ahah, sorry I missed it in the other one!

Reply:
we get a bunch more info from the server about the background job (see `IFileInfo`). not strictly necessary for the feature as it stands in this PR, but we're definitely gonna want to show that info soon

also, the established pattern in the frontend is to load fresh data from the server after updating it, so i was mostly following that pattern

Reply:
That's what my instincts were saying, but I wanted to defer to you since you wrote the code. :slightly_smiling_face: